iquip/apps/dataviewer.py

```diff
@@ -762,11 +762,11 @@ def setData(self, data: np.ndarray, axes: Sequence[AxisInfo]):
             logger.error("MainPlotWidget does not support %d-dim data", data.ndim)
             return
         self.viewers[plotType].setData(data, axes)
-        self.stack.setCurrentIndex(plotType)
         if self.autoRangeBox.isChecked():
             plotItem = self.viewers[plotType].plotItem
             bounds = plotItem.getViewBox().childrenBoundingRect(items=None)
             plotItem.setXRange(bounds.left(), bounds.right())
+        self.stack.setCurrentIndex(plotType)
```

Comment:
I think this line is just moved from line 765.
If it is not intended, please move it back to line 765, since it makes an unnecessary diff.

Reply:
As I think, it looks better to show the plot after all modifications are done.
What do you think of it?

Reply:
Oh I agree with that! Let's go with your decision.